controller/controller_test.go

```diff
@@ -270,508 +277,815 @@ func TestUnhealthyIfUpdaterFails(t *testing.T) {
+	runAndAssertUpdates(t, expectGetAllIngresses, testSpec{
+		"ingress without rules definition",
+		createIngressWithoutRules(),
+		createServiceFixture(ingressSvcName, "lalala land", serviceIP),
+		createDefaultNamespaces(),
+		nil,
+		defaultConfig(),
+	})
+}
 
-		config.KubernetesClient = client
-		config.Updaters = []Updater{updater}
+type clientExpectation func(client *fake.FakeClient, ingresses []*v1beta1.Ingress)
 
-		controller := New(config)
+var expectGetAllIngresses = func(client *fake.FakeClient, ingresses []*v1beta1.Ingress) {
+	client.On("GetAllIngresses").Return(ingresses, nil)
+}
 
-		updater.On("Start").Return(nil)
-		updater.On("Stop").Return(nil)
-		// once for ingress update, once for service update
-		updater.On("Update", test.entries).Return(nil).Times(2)
+var expectGetIngresses = func(client *fake.FakeClient, ingresses []*v1beta1.Ingress) {
+	client.On("GetIngresses", &k8s.NamespaceSelector{LabelName: "team", LabelValue: "theteam"}).Return(ingresses, nil)
+}
 
-		client.On("GetIngresses").Return(test.ingresses, nil)
-		client.On("GetServices").Return(test.services, nil)
+func runAndAssertUpdates(t *testing.T, clientExpectation clientExpectation, test testSpec) {
+	//given
+	assert := assert.New(t)
+
+	fmt.Printf("test: %s\n", test.description)
+	// add ingress pointers to entries
+	test.entries = addIngresses(test.ingresses, test.entries)
 
-		ingressWatcher, ingressCh := createFakeWatcher()
-		serviceWatcher, serviceCh := createFakeWatcher()
-		client.On("WatchIngresses").Return(ingressWatcher)
-		client.On("WatchServices").Return(serviceWatcher)
+	// setup clients
+	client := new(fake.FakeClient)
+	updater := new(fakeUpdater)
 
-		//when
-		assert.NoError(controller.Start())
-		ingressCh <- struct{}{}
-		serviceCh <- struct{}{}
-		time.Sleep(smallWaitTime)
+	config := test.config
 
-		//then
-		assert.NoError(controller.Stop())
-		time.Sleep(smallWaitTime)
-		updater.AssertExpectations(t)
-	}
+	config.KubernetesClient = client
+	config.Updaters = []Updater{updater}
+
+	controller := New(config)
+
+	updater.On("Start").Return(nil)
+	updater.On("Stop").Return(nil)
+	// once for each update: ingress, service, namespace
+	updater.On("Update", test.entries).Return(nil).Times(3)
+
+	clientExpectation(client, test.ingresses)
+	client.On("GetServices").Return(test.services, nil)
+
+	ingressWatcher, ingressCh := createFakeWatcher()
+	serviceWatcher, serviceCh := createFakeWatcher()
+	namespaceWatcher, namespaceCh := createFakeWatcher()
+	client.On("WatchIngresses").Return(ingressWatcher)
+	client.On("WatchServices").Return(serviceWatcher)
+	client.On("WatchNamespaces").Return(namespaceWatcher)
+
+	//when
+	assert.NoError(controller.Start())
+	ingressCh <- struct{}{}
+	serviceCh <- struct{}{}
+	namespaceCh <- struct{}{}
+	time.Sleep(smallWaitTime)
+
+	//then
+	assert.NoError(controller.Stop())
+	time.Sleep(smallWaitTime)
+	updater.AssertExpectations(t)
+	client.AssertExpectations(t)
```

Comment:
Assert on the `client` calls, to make sure it's receiving the correct `GetAllIngresses`/`GetIngresses(namespaceSelector)` call.